### PR TITLE
docs: add pnc component introduction

### PR DESCRIPTION
**Pull request by JackFu123**

### CODE REVIEW

File: docs/technical_documents/planning_component.md [hunk 1]
```diff
@@ -0,0 +1,33 @@
...
+## How the messages are received and sent
+![Diagram](images/planning_component_api.png)
+
+1. Input
+Messages are partially input from readers, which are defined in PlanningComponent::Init(). For Planning component, readers receive message types of RoutingResponse/TrafficLightDetection/PadMessage/Stories/MapMsg and store them in local for processing afterward.
+Another part of input messages are defined in PlanningComponent::Proc(), where the planning process is called when new messages arrives.
+2. Process. PlanningComponent::Proc() function works like Reader's callback function. Each time there are new messages of prediction::PredictionObstacles/canbus::Chassis/localization::LocalizationEstimate, this process is implemented till the end of this function.
+3. Output. PlanningComponent output the message by the following line `planning_writer_->Write(adc_trajectory_pb)`, in which the writer send the final adc_trajectory_pb message to the cyber system. The writer is initialized in PlanningComponent::Init(), just after the readers are initialized.
+
+## How the trajectory is arranged and made up
```
Comment (JasonZhou404) [inline docs/technical_documents/planning_component.md:30]: "How the trajectory as planning component output is produced" ?

File: docs/technical_documents/planning_component.md [hunk 2]
```diff
@@ -0,0 +1,33 @@
+# PnC Component Introduction
```
Comment (JasonZhou404) [inline docs/technical_documents/planning_component.md:1]: Planning component?

File: docs/technical_documents/planning_component.md [hunk 3]
```diff
@@ -0,0 +1,33 @@
...
+Another part of input messages are defined in PlanningComponent::Proc(), where the planning process is called when new messages arrives.
+2. Process. PlanningComponent::Proc() function works like Reader's callback function. Each time there are new messages of prediction::PredictionObstacles/canbus::Chassis/localization::LocalizationEstimate, this process is implemented till the end of this function.
+3. Output. PlanningComponent output the message by the following line `planning_writer_->Write(adc_trajectory_pb)`, in which the writer send the final adc_trajectory_pb message to the cyber system. The writer is initialized in PlanningComponent::Init(), just after the readers are initialized.
+
+## How the trajectory is arranged and made up
+Now we know how messages are input to and output from the planning component. The work of planning module is basically processing the input messages and arranging an appropriate trajectory for the ADC. The final goal of this is to ensure safety, comfort and passability. And different algrithms are explored in Apollo in order to achieve or get closer to this goal.
```
Comment (JasonZhou404) [inline docs/technical_documents/planning_component.md:31]: arrange? maybe "produce"?

**Review by JasonZhou404 — CHANGES_REQUESTED** [review]

File: docs/technical_documents/planning_component.md (hunk 3 above)
Comment (spicyog) [inline docs/technical_documents/planning_component.md:31]: algorithms

File: docs/technical_documents/planning_component.md [hunk 4]
```diff
@@ -0,0 +1,33 @@
...
+- The basic and more mature alogrithm is rule-based planning, where we use a two-layer state machine to manage the path in different scenarios. For details, please refer [scenarios](https://github.com/ApolloAuto/apollo/tree/master/modules/planning/conf/scenario).
```
Comment (spicyog) [inline docs/technical_documents/planning_component.md:32]: algorithms

File: docs/technical_documents/planning_component.md (hunk 2 above)
Comment (JackFu123) [reply docs/technical_documents/planning_component.md]: changed to "Planning component"

File: docs/technical_documents/planning_component.md (hunk 1 above)
Comment (JackFu123) [reply docs/technical_documents/planning_component.md]: Changed to "How the trajectory as planning component output is produced"

File: docs/technical_documents/planning_component.md (hunk 3 above)
Comment (JackFu123) [reply docs/technical_documents/planning_component.md]: Changed arranging to producing

Comment (JackFu123) [reply docs/technical_documents/planning_component.md]: corrected the typos

**Review by JasonZhou404 — APPROVED** [review]

File: docs/technical_documents/planning_component.md (hunk 4 above)
Comment (JackFu123) [reply docs/technical_documents/planning_component.md]: corrected the typos

**Pull request merged**